### DISCUSSION TITLE
Serial on tinyAVR 2-series (1624)

### DISCUSSION THREAD

**Discussion by gcormier** [SpenceKonde/megaTinyCore#393] [Q&A]
So... can anyone get it working? I have a simple sketch now.

```
void setup() {
  Serial1.begin(19200);
}

void loop() {
  Serial1.println("Hello");
  
  delay(1000);
}
```

And I'm not getting anything from Serial1, which would be PA1/PA2 on the 1624.

**Comment by jvasileff** [comment]
FWIW, that program works for me on a 1626 (I don't have a 1624), with TX on PA1.

**Reply by gcormier** [reply]
Are you using the internal oscillator and at what speed?

**Reply by jvasileff** [reply]
Internal with megaTinyCore defaults, so 20MHz I believe.

**Comment by SpenceKonde** [comment]
Will test this when I get back from the store.

Edited title of thread to avoid the ambiguous term "avr2" (ithe only place I've seen that all-lowercase-no-space-number is in the compiler specs for various classic AVRs ((arduino install location)\hardware\tools\avr\avr\lib ) but no avr2 in that list)

**Comment by SpenceKonde** [comment]
works for me with both 1624 and 1626 *shrug*

**Comment by gcormier** [comment]
k lemme bust out the scope, maybe the serial adapter i have is not working with the 1624 for some reason

**Reply by SpenceKonde** [reply]
Do you have TX and RX right way around? First thing I test with serial is always swapping TX and RX ;-)

**Reply by gcormier** [reply]
Did that already :)

**Reply by WaynePauly** [reply]
I have the same issue on a 3224- Serial1.begin(9600) ; does not seem to initialize the Serial1 UART, Rx seems to be an output. Are there are any extra steps needed in programming it? PA2 IS the Rx pin- right?

**Comment by WaynePauly** [comment]
My bad- it was a cold SMD solder joint. Serial1 works fine now, no programming changes other than 10 Mhz internal osc.
